File: jsfring-webservice/src/main/java/fr/pinguet62/jsfring/webservice/converter/RightMapper.java
package fr.pinguet62.jsfring.webservice.converter;

import fr.pinguet62.jsfring.model.sql.Right;
import fr.pinguet62.jsfring.webservice.config.MapstructConfig;
import fr.pinguet62.jsfring.webservice.dto.RightDto;
import org.mapstruct.Mapper;
import org.mapstruct.MappingTarget;

import java.util.List;

@Mapper(config = MapstructConfig.class)
public interface RightMapper {

    RightDto toDto(Right entity);

    List<RightDto> toDto(List<Right> entities);

    void updateFromDto(Right entity, @MappingTarget RightDto dto);

}

File: jsfring-webservice/src/main/kotlin/fr/pinguet62/jsfring/webservice/converter/RightMapper.kt
package fr.pinguet62.jsfring.webservice.converter

import fr.pinguet62.jsfring.model.sql.Right
import fr.pinguet62.jsfring.webservice.config.MapstructConfig
import fr.pinguet62.jsfring.webservice.dto.RightDto
import org.mapstruct.Mapper
import org.mapstruct.MappingTarget

@Mapper(config = MapstructConfig::class)
interface RightMapper {

    fun toDto(entity: Right): RightDto

    fun toDto(entities: List<Right>): List<RightDto>

    fun updateFromDto(entity: Right, @MappingTarget dto: RightDto)

}
